dask/base.py

```diff
@@ -98,6 +102,48 @@ def _restore_excepthook():
 original_excepthook = sys.excepthook
 sys.excepthook = _clean_traceback_hook(sys.excepthook)
 
+_annotations: ContextVar[dict] = ContextVar("annotations", default={})
+
+
+def get_annotations(default_value: Any = None) -> dict | None:
+    """Get global annotations.
+
+    Parameters
+    ----------
+    default_value: Any
+        What to return if no annotations are set
+
+    Returns
+    -------
+    result : dict | None
+        Dict of annotations, if any
+    """
+    return _annotations.get() or default_value
+
+
+def get_annotation(key: str, *, default_value: Any = None) -> Any:
```

Comment:
Is this second function truly necessary? If so, we should choose different function names. The `s` is very easy to miss

Reply:
Not necessary, but it makes for prettier code in `distributed` here:

https://github.com/dask/distributed/blob/1a489575abe7bfbe4459212e44f47bf9dc962487/distributed/spans.py#L77

I can remove it though.